### DISCUSSION TITLE
Using GitHub actions

### DISCUSSION THREAD

**Discussion by shivan-s** [mrdbourke/nutrify#14] [Ideas]
I heard something about CI/CD.

Github actions is good for this.

I have no idea how it works.

So I'm keen to tag along with this.

Edit: The idea is you commit your code and this will automatically update the production code. No downtime, all in one automated step.

**Comment by mrdbourke** [comment]
GitHub Actions works great, I use it to build [learntensorflow.io](https://learntensorflow.io) from the [`tensorflow-deep-learning`](https://github.com/mrdbourke/tensorflow-deep-learning) repo.

See `.github/workflows`, essentially a `.yaml` file with instructions to build the site.

We'll keep this in mind for future features of Nutrify - right now, Streamlit Cloud builds and deploys it automatically, but if we ever have to move off Streamlit to custom hosting (likely in the future), GitHub Actions is probably what we'll use.

**Reply by shivan-s** [reply]
Awesome. Thanks for that.
